aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -186,27 +186,6 @@ void bitwise_xor_kernel(TensorIterator& iter) {
   }
 }
 
-template<typename scalar_t>
-static inline scalar_t lshift_wrapper(scalar_t a, scalar_t b) {
-  return a << b;
-}
-
-static inline int8_t lshift_wrapper(int8_t a, int8_t b) {
-  return ((uint8_t)a) << b;
-}
-
-static inline int16_t lshift_wrapper(int16_t a, int16_t b) {
-  return ((uint16_t)a) << b;
-}
-
-static inline int32_t lshift_wrapper(int32_t a, int32_t b) {
-  return ((uint32_t)a) << b;
-}
-
-static inline int64_t lshift_wrapper(int64_t a, int64_t b) {
-  return ((uint64_t)a) << b;
```

Comment:
Just out of curiosity, why do we have to make the left hand side unsigned before performing the shift?

Reply:
Probably because shifting on signed integers in C++ is implementation defined. See https://stackoverflow.com/a/4009954/1150462 for an explanation and reference.